php-nginx/whitelist_functions.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+function filter_blacklist($val)
+{
+    global $whitelist;
+    return !in_array($val, $whitelist);
+}
+
+$phpDir = getenv("PHP_DIR");
+
+if ($phpDir === false) {
+    exit(0);
+}
+
+$whitelist = getenv("WHITELIST_FUNCTIONS");
+
+if ($whitelist === false) {
+    exit(0);
+} else {
+    $whitelist = array_filter(array_map('trim', explode(',', $whitelist)));
+}
+
+$suhosinBlacklist = ini_get('suhosin.executor.func.blacklist');
+$disable_functions = ini_get('disable_functions');
+
+$whitelist_ini = '';
+
+if ($suhosinBlacklist !== false) {
+    $suhosinBlacklist = array_filter(
+        array_map('trim', explode(',', $suhosinBlacklist)));
```

Comment:
just use `array_diff` here instead of defining a custom function, global variables, and using array_map, i.e:

```php
$whitelist_ini .= sprintf(
         "suhosin.executor.func.blacklist=\"%s\"\n",
         implode(
             ',',
             array_diff($suhosinBlacklist, $whitelist)
         )
     );
```